src/vsr/replica.zig

```diff
@@ -445,6 +446,9 @@ pub fn ReplicaType(
         /// protocol.
         do_view_change_quorum: bool = false,
 
+        checkpoint_from_all_replicas: CheckpointQuorum = .{},
+        checkpoint_quorum: bool = false,
```

Comment:
Document the menaning of the field. While `checkpoint_from_all_replicas` is somewhat obivous, this bool isn't at all.

Reply:
I am actually wondering, do we need to maintain it as a field? Can we make this a function that looks at self.checkpoint_from_all_replicas, self.solo() and self.op() ?

Reply:
Would be quite clean to make it a function of these, but I would have to give it some deeper thought if this is possible...

As of now, whether we mark self.checkpoint_quorum as True seems quite dynamic; we try to use incoming pings, commits, and DVC messages (in case we haven't received enough pings/commits and we don't want to wait to hear about a checkpoint quorum through them).

Reply:
Took a stab at this as part of the latest commit: https://github.com/tigerbeetle/tigerbeetle/pull/2104/files#diff-e95e4dc6c1aae349e798cfeed22ad01f830cce0e654e6b4b765b3e7b0c677b6aR5856 😄